### DISCUSSION TITLE
onFail function when no tenant is found for a given domain/subdomain.

### DISCUSSION THREAD

**Discussion by andelwebsolutions** [spatie/laravel-multitenancy#390] [Ideas]
Hi, I would like to request a feature to redirect to a "main" domain/website (a marketing website for example) if no tenant is found for a given (sub-)domain.

Currenlty this is not possible I believe. Preferably this would be a "onFail" function that we could implement on our TenantFinder class to redirect the user (or do something else).

**Comment by kassah** [comment]
You can achieve this behavior by registering a renderable for the NoCurrentTenant in your \App\Exceptions\Handler class.

```php
// in app/Exceptions/Handler.php
// ...
use Spatie\Multitenancy\Exceptions\NoCurrentTenant;
use Illuminate\Support\Facades\Redirect;
// ...
    public function register()
    {
        // ...
        $this->renderable(function (NoCurrentTenant $nte) {
            return Redirect::away('https://www.mymainwebsite.com/');
        });
    }
// ...
```

See this [phpsandbox file for a demonstration of how that's done](https://phpsandbox.io/e/x/notenantresponsedemo-g2umz?layout=EditorPreview&defaultPath=%2F&theme=dark&showExplorer=no&openedFiles=%2Fapp%2FExceptions%2FHandler.php&files=%2Fapp%2FExceptions%2FHandler.php). I returned a view there since I can't redirect an iframe, a limitation of phpsandbox.io, not Laravel.
